sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py

```diff
@@ -771,19 +763,19 @@ def get_blob_client(
                 "Please use 'BlobProperties.name' or any other str input type instead.",
                 DeprecationWarning
             )
-        try:
+        if isinstance(container, ContainerProperties):
             container_name = container.name
-        except AttributeError:
+        else:
             container_name = container
-        try:
+        if hasattr(blob, 'name'):
```

Comment:
Seems like this is probably here for `BlobPropeties`? (which we deprecated but still need to support). Should we do `isinstance(blob, BlobProperties` instead?

Reply:
Instead of having two `isinstance(blob, BlobProperties)` checks, I just combined it with the one above (that was used to just throw the warning) and also moved the else statement (non-deprecated case) to function as expected.